kernel/src/scan/mod.rs

```diff
@@ -199,11 +199,13 @@ impl Scan {
         let commit_read_schema = get_log_schema().project(&[ADD_NAME, REMOVE_NAME])?;
         let checkpoint_read_schema = get_log_schema().project(&[ADD_NAME])?;
 
+        // NOTE: We don't pass any meta-predicate because we expect no meaningful row group skipping
+        // when ~every checkpoint file will contain the adds and removes we are looking for.
         let log_iter = self.snapshot.log_segment.replay(
             engine,
             commit_read_schema,
             checkpoint_read_schema,
-            self.predicate.clone(),
```

Comment:
NOTE: This was an existing bug -- passing a query-level filter to the metadata file reads.

Reply:
opened https://github.com/delta-incubator/delta-kernel-rs/issues/380

Reply:
Fixed by new `replay_for_scan_data` test in https://github.com/delta-incubator/delta-kernel-rs/pull/381

Reply:
ah sorry! thanks!!